Add hover

ericdallo opened the pull request:
* Add hover
* Refactor to return a nubank function with the packages.

Note: I'll bump hover in a next PR

ericdallo committed:
Add nubank prefix

ericdallo committed:
Add hover

thiagokokada commented:
`nubank.hover` is not working after adding in `environment.systemPackages`:

```
error: The option value `environment.systemPackages.[definition 1-entry 18]' in `/etc/nixos/nubank-configuration.nix' is not of type `package'.
```

ericdallo committed:
Fix hover call

thiagokokada commented:
Hover is generating non-static binaries, at least in NixOS stable:

```bash
$ hover run 
hover: Missing config: file hover.yaml not found: open go/hover.yaml: no such file or directory
hover: ⚠ The go-flutter project tries to stay compatible with the beta channel of Flutter.
hover: ⚠     It's advised to use the beta channel: `flutter channel beta`
hover: Building flutter bundle
Font subetting is not supported in debug mode. The --tree-shake-icons flag will be ignored.
hover: Using engine from cache
hover: Compiling go binary using docker container
Running this docker command: /nix/store/3vj9y663i3l0avp7bdn3pbn671k4mxgq-docker-19.03.12/bin/docker run --rm --mount type=bind,source=/home/thiagoko/dev/nu/mini-meta-repo,target=/app --mount type=bind,source=/home/thiagoko/.cache/hover/engine,target=/root/.cache/hover/engine --mount type=bind,source=/home/thiagoko/.cache/hover/docker-go-cache,target=/go-cache --env GOCACHE=/go-cache --env HOVER_SAFE_CHOWN_UID=1000 --env HOVER_SAFE_CHOWN_GID=100 --env GOPROXY=https://proxy.golang.org,direct
 --env GOPRIVATE=
 --env HOVER_IN_DOCKER_BUILD_VMARGS=--observatory-port=50300,--enable-service-port-fallback,--disable-service-auth-codes goflutter/hover:latest hover-safe.sh build linux --skip-flutter-build-bundle --skip-engine-download --debug
docker container: hover: Missing config: file hover.yaml not found: open go/hover.yaml: no such file or directory
docker container: hover: Compiling 'go-flutter' and plugins
docker container: go: downloading github.com/lucasmafra/go-flutter-share v0.0.0-20200213183323-6ca7e6570bca
docker container: go: downloading github.com/go-flutter-desktop/go-flutter v0.41.2
docker container: go: downloading github.com/go-flutter-desktop/plugins/shared_preferences v0.4.0
docker container: go: extracting github.com/lucasmafra/go-flutter-share v0.0.0-20200213183323-6ca7e6570bca
docker container: go: extracting github.com/go-flutter-desktop/plugins/shared_preferences v0.4.0
docker container: go: downloading github.com/syndtr/goleveldb v1.0.0
docker container: go: downloading github.com/mitchellh/go-homedir v1.1.0
docker container: go: extracting github.com/mitchellh/go-homedir v1.1.0
docker container: go: extracting github.com/go-flutter-desktop/go-flutter v0.41.2
docker container: go: extracting github.com/syndtr/goleveldb v1.0.0
docker container: go: downloading github.com/Xuanwo/go-locale v0.3.0
docker container: go: downloading golang.org/x/text v0.3.3
docker container: go: downloading github.com/go-gl/glfw v0.0.0-20191125211704-12ad95a8df72
docker container: go: downloading github.com/go-gl/gl v0.0.0-20190320180904-bf2b1f2f34d7
docker container: go: extracting github.com/Xuanwo/go-locale v0.3.0
docker container: go: downloading github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
docker container: go: extracting github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
docker container: go: extracting github.com/go-gl/glfw v0.0.0-20191125211704-12ad95a8df72
docker container: go: downloading github.com/go-gl/glfw/v3.3/glfw v0.0.0-20200707082815-5321531c36a2
docker container: go: extracting github.com/go-gl/glfw/v3.3/glfw v0.0.0-20200707082815-5321531c36a2
docker container: go: extracting github.com/go-gl/gl v0.0.0-20190320180904-bf2b1f2f34d7
docker container: go: extracting golang.org/x/text v0.3.3
docker container: go: finding github.com/lucasmafra/go-flutter-share v0.0.0-20200213183323-6ca7e6570bca
docker container: go: finding github.com/go-flutter-desktop/plugins/shared_preferences v0.4.0
docker container: go: finding github.com/go-flutter-desktop/go-flutter v0.41.2
docker container: go: finding github.com/mitchellh/go-homedir v1.1.0
docker container: go: finding github.com/Xuanwo/go-locale v0.3.0
docker container: go: finding github.com/syndtr/goleveldb v1.0.0
docker container: go: finding github.com/go-gl/glfw/v3.3/glfw v0.0.0-20200707082815-5321531c36a2
docker container: go: finding golang.org/x/text v0.3.3
docker container: go: finding github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
docker container: go: finding github.com/go-gl/gl v0.0.0-20190320180904-bf2b1f2f34d7
docker container: hover: Successfully compiled executable binary for linux
hover: Docker run completed
hover: Build finished, starting app...
hover: Running nubank in debug mode
hover: Failed to start app 'nubank': fork/exec ./go/build/outputs/linux/nubank: no such file or directory

$ ldd ./go/build/outputs/linux/nubank
	linux-vdso.so.1 (0x00007ffc791f9000)
	libGL.so.1 => not found
	libX11.so.6 => not found
	libXrandr.so.2 => not found
	libXxf86vm.so.1 => not found
	libXi.so.6 => not found
	libXcursor.so.1 => not found
	libm.so.6 => /nix/store/xg6ilb9g9zhi2zg1dpi4zcp288rhnvns-glibc-2.30/lib/libm.so.6 (0x00007f4ea69fa000)
	libXinerama.so.1 => not found
	libdl.so.2 => /nix/store/xg6ilb9g9zhi2zg1dpi4zcp288rhnvns-glibc-2.30/lib/libdl.so.2 (0x00007f4ea69f5000)
	librt.so.1 => /nix/store/xg6ilb9g9zhi2zg1dpi4zcp288rhnvns-glibc-2.30/lib/librt.so.1 (0x00007f4ea69eb000)
	libflutter_engine.so => /home/thiagoko/dev/nu/mini-meta-repo/./go/build/outputs/linux/libflutter_engine.so (0x00007f4ea4c78000)
	libpthread.so.0 => /nix/store/xg6ilb9g9zhi2zg1dpi4zcp288rhnvns-glibc-2.30/lib/libpthread.so.0 (0x00007f4ea4c55000)
	libc.so.6 => /nix/store/xg6ilb9g9zhi2zg1dpi4zcp288rhnvns-glibc-2.30/lib/libc.so.6 (0x00007f4ea4a96000)
	/lib64/ld-linux-x86-64.so.2 => /nix/store/xg6ilb9g9zhi2zg1dpi4zcp288rhnvns-glibc-2.30/lib64/ld-linux-x86-64.so.2 (0x00007f4ea6b3e000)
```

ericdallo commented:
@thiagokokada Closing as I added a patch and bumped hover on [this PR](https://github.com/nubank/nixpkgs/pull/4).
Added a patch for building with `CGO_ENABLED=1` but `ldd` shows the same output:
![image](https://user-images.githubusercontent.com/7820865/91627431-a1a7a300-e98d-11ea-9c86-614db24de39f.png)
